File: ReverseString.java
/*
https://leetcode.com/problems/reverse-string/

Write a function that reverses a string. The input string is given as an array of characters s.
You must do this by modifying the input array in-place with O(1) extra memory.
*/

class Solution {
    public void reverseString(char[] s) {
        int left = 0, right = s.length-1;
        for(int i=0;i<s.length;i++){
            char temp = s[left];
            s[left++] = s[right];
            s[right--] = temp;
            if(left >= right) break;
        }
    }
}

File: ReverseString.py
#Two pointers
class solution:
  def reverseString(self, s: List[str]) -> None:
    left, right = 0, len(s) -1
    
    while left < right:
      s[left], s[right] = s[right], s[left]
      left += 1
      right += 1
      
 #Pythonic way
class solution:
  def reverseString(self, s: List[str]) -> None:
    s[:] = s[::-1] #Leetcode dosen't allow to use extra space
    # Or s.reverse()
